server/application/src/main/kotlin/fr/devlille/partners/connect/companies/application/CompanyJobOfferPromotionRepositoryExposed.kt

```diff
@@ -0,0 +1,166 @@
+package fr.devlille.partners.connect.companies.application
+
+import fr.devlille.partners.connect.companies.domain.CompanyJobOfferPromotionRepository
+import fr.devlille.partners.connect.companies.domain.EmbeddedJobOffer
+import fr.devlille.partners.connect.companies.domain.JobOfferPromotionResponse
+import fr.devlille.partners.connect.companies.infrastructure.db.CompanyEntity
+import fr.devlille.partners.connect.companies.infrastructure.db.CompanyJobOfferEntity
+import fr.devlille.partners.connect.companies.infrastructure.db.CompanyJobOfferPromotionEntity
+import fr.devlille.partners.connect.companies.infrastructure.db.CompanyJobOfferPromotionsTable
+import fr.devlille.partners.connect.internal.infrastructure.api.ConflictException
+import fr.devlille.partners.connect.internal.infrastructure.api.ForbiddenException
+import fr.devlille.partners.connect.internal.infrastructure.api.PaginatedResponse
+import fr.devlille.partners.connect.internal.infrastructure.api.paginated
+import fr.devlille.partners.connect.internal.infrastructure.api.toPaginatedResponse
+import fr.devlille.partners.connect.internal.infrastructure.db.PromotionStatus
+import fr.devlille.partners.connect.partnership.infrastructure.db.PartnershipEntity
+import io.ktor.server.plugins.NotFoundException
+import kotlinx.datetime.Clock
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.toLocalDateTime
+import org.jetbrains.exposed.v1.core.SortOrder
+import org.jetbrains.exposed.v1.core.SqlExpressionBuilder.eq
+import org.jetbrains.exposed.v1.core.and
+import org.jetbrains.exposed.v1.jdbc.transactions.transaction
+import java.util.UUID
+
+/**
+ * Exposed-based implementation of CompanyJobOfferPromotionRepository.
+ *
+ * Handles job offer promotion operations from the company perspective, including
+ * validation of business rules, upsert logic for declined promotions, and
+ * pagination of promotion lists.
+ */
+class CompanyJobOfferPromotionRepositoryExposed : CompanyJobOfferPromotionRepository {
+    override fun promoteJobOffer(
+        companyId: UUID,
+        partnershipId: UUID,
+        jobOfferId: UUID,
+    ): UUID = transaction {
+        // FR-001: Validate company owns the job offer
+        val company = CompanyEntity.findById(companyId)
+            ?: throw NotFoundException("Company not found: $companyId")
+
+        val jobOffer = CompanyJobOfferEntity.findById(jobOfferId)
+            ?: throw NotFoundException("Job offer not found: $jobOfferId")
+
+        if (jobOffer.company.id.value != companyId) {
+            throw ForbiddenException("Job offer $jobOfferId does not belong to company $companyId")
```

Comment:
This can be simplified by only one query and return a single 404 if there isn't any job offer in the company.